chore: write WebSocket service README (#7)
virtool/virtool-ws: README.md

poetry python run.py 
```

## Contributing

### Commits

All commits must follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0) specification.

These standardized commit messages are used to automatically publish releases using [`semantic-release`](https://semantic-release.gitbook.io/semantic-release)
after commits are merged to `main` from successful PRs.

**Example**

```text
feat: add API support for assigning labels to existing samples
```

Descriptive bodies and footers are required where necessary to describe the impact of the commit. Use bullets where appropriate.

Additional Requirements

1. **Write in the imperative**. For example, _"fix bug"_, not _"fixed bug"_ or _"fixes bug"_.
2. **Don't refer to issues or code reviews**. For example, don't write something like this: _"make style changes requested in review"_.
   Instead, _"update styles to improve accessibility"_.
3. **Commits are not your personal journal**. For example, don't write something like this: _"got server running again"_
   or _"oops. fixed my code smell"_.

From Tim Pope: [A Note About Git Commit Messages](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html)

> ### Tests

Reviewer: Move this above **Commits**.

```

## Contributing

### Commits

All commits must follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0) specification.

These standardized commit messages are used to automatically publish releases using [`semantic-release`](https://semantic-release.gitbook.io/semantic-release)
after commits are merged to `main` from successful PRs.

**Example**

```text
feat: add API support for assigning labels to existing samples
```

Descriptive bodies and footers are required where necessary to describe the impact of the commit. Use bullets where appropriate.

Additional Requirements

1. **Write in the imperative**. For example, _"fix bug"_, not _"fixed bug"_ or _"fixes bug"_.
2. **Don't refer to issues or code reviews**. For example, don't write something like this: _"make style changes requested in review"_.
   Instead, _"update styles to improve accessibility"_.
3. **Commits are not your personal journal**. For example, don't write something like this: _"got server running again"_
   or _"oops. fixed my code smell"_.

From Tim Pope: [A Note About Git Commit Messages](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html)

### Poetry

> Dependencies & virtual environments are managed with [Poetry](https://python-poetry.org/ "Poetry")

Reviewer: ```suggestion
Dependencies & virtual environments are managed with [Poetry](https://python-poetry.org).
```

To install `poetry`:

```sh
sudo pip install poetry
```

> 

Reviewer: Remove this. It is not the recommended way to install Poetry.

https://python-poetry.org/docs/master/#installing-with-the-official-installer

Add a line like: "Poetry can be installed by following the [official installation guide](https://python-poetry.org/docs/master/#installing-with-the-official-installer)".

> To install dependencies, and the `virtool-ws` package, into a virtual environment:

Reviewer: ```suggestion
To install dependencies and `virtool-ws` install a virtual environment, clone the repository and:
```

```sh

> git clone https://github.com/virtool/virtool-ws
> cd virtool-ws
> 

Reviewer: ```suggestion
```

poetry install
```

To run commands in the virtual environment:

```sh
poetry run <<command>>
```

### Tests

> [Pytest](https://docs.pytest.org/en/7.1.x/ "Pytest") is used to implement unit

Reviewer: ```suggestion
[Pytest](https://docs.pytest.org) is used to implement unit
```

and integration tests.

`virtool-ws` depends on [Redis](https://redis.io/) 
to be running so [docker-compose](https://docs.docker.com/compose/) 
will need to be installed on your system for this to work. It might also 
be necessary to set up a `docker` user group on your system, so you can 
[use docker without sudo](https://linoxide.com/use-docker-without-sudo-ubuntu/).

The Docker Compose configuration for `virtool-ws` development along with the guide for adding it to your local machine is provided
at this [repository](https://github.com/virtool/compose). After you have made the necessary changes, start 
the services:

```shell
cd Projects/compose 
docker compose -p virtool --profile test up -d

> ```

Reviewer: You don't need to document basics of using Docker and Compose. Replace all of this with:

> The test suite requires Redis to be running and configured. See the [Virtool compose repository](...) for a predefined configuration that can be used for developing Virtool services.

To run the entire test suite:

```sh

> cd virtool-ws
> pytest

Reviewer: ```suggestion
poetry run pytest
```
